src/OrchardCore/OrchardCore/Shell/ShellHost.cs

```diff
@@ -378,6 +378,7 @@ private Task<ShellContext> CreateSetupContextAsync(ShellSettings defaultSettings
                 // Creates a default shell settings based on the configuration.
                 var shellSettings = _shellSettingsManager.CreateDefaultSettings();
                 shellSettings.Name = ShellHelper.DefaultShellName;
+                shellSettings.VersionId = IdGenerator.GenerateId();
```

Comment:
Will this generate a new Id every time I go to the setup page from, let's say, different browsers?

Reply:
`CreateSetupContextAsync()` is only used for the Default tenant.

For all other tenants, each time we create/update settings, the settings are persisted so you will retrieve the same Id from any browser and any app app instance if the settings store is shared.

Here for the default tenant, the settings for setup are only set in memory, so you will get the same Id from any browser if you target the same app instance. The Id will change if you restart the app instance or target another one.

But it is so simple to also call here `UpdateShellSettingsAsync()` which will set once a new VersionId, as for other tenants, I did the update and tried it, now you will get the same Id from any browser and any targetted instance, also for the Default tenant.